tools/scripts/generate.py

```diff
@@ -0,0 +1,113 @@
+import requests
+import argparse
+import yaml
+import os
+from pathlib import Path
+from slugify import slugify
+
+cwd = os.getcwd()
+DATA_PATH = os.path.join(cwd, 'data')
+
+BUGZILLA_URL = "https://bugzilla.mozilla.org/show_bug.cgi?id="
+BUGZILLA_API = "https://bugzilla.mozilla.org/rest/bug/"
+
+BREAKAGE_STR = ["webcompat", "mozilla-mobile"]
+PLATFORM_STR = ["bugs.chromium.org", "bugs.webkit.org"]
+
+YAML_FILE_EXT = ("*.yaml", "*.yml")
+
+
+def fetch_bug_by_id(bug_id) -> dict:
+    url = BUGZILLA_API + bug_id
+    params = {"include_fields": "summary,see_also"}
+    response = requests.get(url, params=params)
+    response.raise_for_status()
+    result = response.json()
+    return result["bugs"][0]
+
+
+def split_see_also_by_type(see_also_list, bug_id) -> list:
+    lists = {
+        "breakage": [f"{BUGZILLA_URL}{bug_id}#c0"],
+        "platform": [BUGZILLA_URL + bug_id]
+    }
+
+    for item in see_also_list:
+        if any(bs in item for bs in BREAKAGE_STR):
+            lists["breakage"].append(item)
+
+        if any(ps in item for ps in PLATFORM_STR):
+            lists["platform"].append(item)
+
+    return lists
+
+
+def build_obj(bug_id) -> dict:
+    bug = fetch_bug_by_id(bug_id)
+    lists = split_see_also_by_type(bug["see_also"], bug_id)
+    data = {
+        "title": bug["summary"],
+        "references": {
+            "breakage": lists["breakage"],
+            "platform_issues": lists["platform"]
+        }
+    }
+
+    return data
+
+
+def check_existing_entries(bug_id) -> list:
+    folder = Path(DATA_PATH)
+    files = [f for f in folder.iterdir() if any(f.match(p) for p in YAML_FILE_EXT)]
+
+    for file in files:
+        with open(file, 'r') as stream:
+            try:
+                contents = yaml.safe_load(stream)
+                # Check the first item in references -> platform_issues list to see if there is a match
+                if "references" in contents and "platform_issues" in contents["references"]:
+                    if contents["references"]["platform_issues"][0] == BUGZILLA_URL + bug_id:
```

Comment:
> which of course won't flag the issue until you run the check, maybe not until pushing to GH

Side note, we could run the checks in a pre-push (or maybe even pre-commit) git hook. Once built, the current `kbcheck` runs in ~0.2s for me, so a git hook would be very reasonable!

Reply:
Yeah a git hook is a good idea. As for the check, we could probably keep it? In case someone uses the script and then fills in the rest of the sections and only after attempting to commit learns that an entry for that bug already exists (so the sooner a duplicate is detected, the better 🙂). I can sync the logic with the kbcheck.